src/repart/repart.c

```diff
@@ -2458,6 +2468,88 @@ static int config_parse_encrypted_volume(
         return 0;
 }
 
+static int config_parse_tpm2_pcrs(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Partition *partition = ASSERT_PTR(data);
+        int r;
+
+        assert(rvalue);
+
+        if (isempty(rvalue)) {
+                /* Clear existing PCR values if empty */
+                partition->tpm2_hash_pcr_values = mfree(partition->tpm2_hash_pcr_values);
+                partition->tpm2_n_hash_pcr_values = 0;
+                return 0;
+        }
+
+        r = tpm2_parse_pcr_argument_append(rvalue, &partition->tpm2_hash_pcr_values, &partition->tpm2_n_hash_pcr_values);
+        if (r < 0) {
+                return 0;
+        }
+
+        return 0;
+}
+
+static int parse_key_file(const char *filename, void **key, size_t *key_size) {
+        _cleanup_(erase_and_freep) char *k = NULL;
+        size_t n = 0;
+        int r;
+
+        r = read_full_file_full(
+                        AT_FDCWD, filename, UINT64_MAX, SIZE_MAX,
+                        READ_FULL_FILE_SECURE|READ_FULL_FILE_WARN_WORLD_READABLE|READ_FULL_FILE_CONNECT_SOCKET,
+                        NULL,
+                        &k, &n);
```

Comment:
or current coding style says that "opaque" func params (i.e. where you pass numeric literals, or niche values and so on) should be prefixed with a comment saying what argument your are passing here.

here the 3rd, 4th, 6th qualify, hence please prefix those arguments with `/* offset=*/`, `/* size= */`, and `/* bind_name= */`.

(nowadays, static analyzers even validate these comments and that they match the parameter names of the func declarations.